### PR TITLE
etcdmain: use /member subdir to save member data

### PR DESCRIPTION
**Pull request by yichengq**

### CODE REVIEW

File: etcdmain/etcd.go [hunk 1]
```diff
@@ -329,6 +330,41 @@ func setupCluster(cfg *config) (*etcdserver.Cluster, error) {
 	return cls, err
 }
 
+func useMemberSubDir(dir string) (string, error) {
```
Comment (xiang90) [inline etcdmain/etcd.go]: createMemberDir(dir string) error

File: etcdmain/etcd.go [hunk 2]
```diff
@@ -329,6 +330,41 @@ func setupCluster(cfg *config) (*etcdserver.Cluster, error) {
...
+		return membdir, nil
+	case !os.IsNotExist(err):
+		return "", err
+	default:
+		// upgrade from data dir without /member
+	}
+	if err := os.MkdirAll(membdir, 0700); err != nil {
+		return "", err
+	}
+	v1NameSet := types.NewUnsafeSet("conf", "log", "snapshot")
```
Comment (xiang90) [inline etcdmain/etcd.go]: v1files

File: etcdmain/etcd.go [hunk 3]
```diff
@@ -329,6 +330,41 @@ func setupCluster(cfg *config) (*etcdserver.Cluster, error) {
...
+	v2NameSet := types.NewUnsafeSet("wal", "snap")
```
Comment (xiang90) [inline etcdmain/etcd.go]: v2files

File: etcdmain/etcd.go [hunk 4]
```diff
@@ -329,6 +330,41 @@ func setupCluster(cfg *config) (*etcdserver.Cluster, error) {
...
 
+func useMemberSubDir(dir string) (string, error) {
+	membdir := path.Join(dir, "member")
+	_, err := os.Stat(membdir)
+	switch {
+	case err == nil:
+		return membdir, nil
+	case !os.IsNotExist(err):
+		return "", err
+	default:
```
Comment (xiang90) [inline etcdmain/etcd.go]: remove the default?

**Comment by yichengq** [comment]
refreshed.

**Comment by xiang90** [comment]
lgtm

**Pull request merged**